sdk/typescript/src/keypairs/passkey/keypair.ts

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { toBase64 } from '@mysten/bcs';
+import { secp256r1 } from '@noble/curves/p256';
+import { blake2b } from '@noble/hashes/blake2b';
+import { randomBytes } from '@noble/hashes/utils';
+import type {
+	AuthenticationCredential,
+	RegistrationCredential,
+} from '@simplewebauthn/typescript-types';
+
+import { PasskeyAuthenticator } from '../../bcs/bcs.js';
+import type { IntentScope, SignatureWithBytes } from '../../cryptography/index.js';
+import { messageWithIntent, SIGNATURE_SCHEME_TO_FLAG, Signer } from '../../cryptography/index.js';
+import type { PublicKey } from '../../cryptography/publickey.js';
+import type { SignatureScheme } from '../../cryptography/signature-scheme.js';
+import {
+	parseDerSPKI,
+	PASSKEY_PUBLIC_KEY_SIZE,
+	PASSKEY_SIGNATURE_SIZE,
+	PasskeyPublicKey,
+} from './publickey.js';
+
+export interface PasskeyCreateOptions {
+	/** The timeout for the passkey creation dialog. */
+	timeout: number;
+	/** The name of the user. */
+	name: string;
+	/** The display name of the user, this is shown at creation dialog. */
+	displayName: string;
+	/** The name of the relying party, this is shown to user in the passkey creation dialog. */
+	rpName: string;
+}
+type DeepPartialConfigKeys = 'rp' | 'user' | 'authenticatorSelection';
+
+type DeepPartial<T> = T extends object
+	? {
+			[P in keyof T]?: DeepPartial<T[P]>;
+		}
+	: T;
+
+type BrowserPasswordProviderOptions = Pick<
+	DeepPartial<PublicKeyCredentialCreationOptions>,
+	DeepPartialConfigKeys
+> &
+	Omit<
+		Partial<PublicKeyCredentialCreationOptions>,
+		DeepPartialConfigKeys | 'pubKeyCredParams' | 'challenge'
+	>;
+
+export interface PasskeyProvider {
+	create(): Promise<RegistrationCredential>;
+	get(challenge: Uint8Array): Promise<AuthenticationCredential>;
+}
+
+// Default browser implementation
+export class BrowserPasskeyProvider implements PasskeyProvider {
+	#name: string;
+	#options: BrowserPasswordProviderOptions;
+
+	constructor(name: string, options: BrowserPasswordProviderOptions) {
+		this.#name = name;
+		this.#options = options;
+	}
+
+	async create(): Promise<RegistrationCredential> {
+		return (await navigator.credentials.create({
+			publicKey: {
+				timeout: 60000,
+				...this.#options,
+				rp: {
+					name: this.#name,
+					...this.#options.rp,
+				},
+				user: {
+					name: this.#name,
+					displayName: this.#name,
+					...this.#options.user,
+					id: randomBytes(10),
+				},
+				challenge: new TextEncoder().encode('Create passkey wallet on Sui'),
+				pubKeyCredParams: [{ alg: -7, type: 'public-key' }],
+				authenticatorSelection: {
+					...this.#options.authenticatorSelection,
+					authenticatorAttachment: 'cross-platform',
```

Comment:
do we want this one to be configurable? This is the less secure version right?

Reply:
on the other hand, this is the more recoverable version...

Reply:
I don't have an opinion here, just wanted to understand if we had a preference one way or the other

Reply:
iiuc, this value will first set by `this.#options.authenticatorSelection`, if its not defined, it will be set by the default as `cross-platform`, no?

Reply:
not as its written. Anything after `this.#options.authenticatorSelection` will be overwritten.

Basically it should be `{ defaultOption: 123, ...userProvidedOptions, optionWeDontWantUserToOverwrite }`

Reply:
got it! i updated to 
```
				authenticatorSelection: {
					authenticatorAttachment: 'cross-platform',
					residentKey: 'required',
					requireResidentKey: true,
					userVerification: 'required',
					...this.#options.authenticatorSelection,
				},
```